plugin-locationcomponent/src/main/java/com/mapbox/maps/plugin/locationcomponent/LocationComponentPluginImpl.kt

```diff
@@ -152,36 +152,38 @@ class LocationComponentPluginImpl : LocationComponentPlugin, LocationConsumer,
 
   private fun activateLocationComponent() {
     if (internalSettings.enabled) {
-      delegateProvider.getStyle { style ->
-        if (locationPuckManager?.isLayerInitialised() == true && isLocationComponentActivated) {
-          return@getStyle
-        }
-        if (locationPuckManager == null) {
-          locationPuckManager = LocationPuckManager(
-            settings = internalSettings,
-            delegateProvider = delegateProvider,
-            positionManager = LocationComponentPositionManager(
-              style,
-              internalSettings.layerAbove,
-              internalSettings.layerBelow
-            ),
-            layerSourceProvider = LayerSourceProvider(),
-            animationManager = PuckAnimatorManager(
-              indicatorPositionChangedListener,
-              indicatorBearingChangedListener
-            )
-          )
-        }
-        locationPuckManager?.let {
-          if (!it.isLayerInitialised()) {
-            it.initialize(style)
-          }
-        }
-        locationPuckManager?.onStart()
-        locationProvider?.registerLocationConsumer(this)
-        isLocationComponentActivated = true
+      delegateProvider.getStyle(::updateStyle)
+    }
+  }
+
+  private fun updateStyle(style: StyleInterface) {
+    if (locationPuckManager?.isLayerInitialised() == true && isLocationComponentActivated) {
```

Comment:
```suggestion
    if (locationPuckManager?.isLayerInitialised() && isLocationComponentActivated) {
```

or 
```suggestion
    if (locationPuckManager?.isLayerInitialised() == true && isLocationComponentActivated == true) {
```

Reply:
First is not possible because first condition could be `null`
Second will work, but Android Studio will notice with message `Simplify boolean expression`

Reply:
ah, okay